src/lib.rs

```diff
@@ -73,12 +73,18 @@ pub mod merkle_proof;
 mod tests;
 pub mod traits;
 pub mod tree;
+#[cfg(feature = "trie")]
+pub mod trie_tree;
 
 #[cfg(feature = "smtc")]
 pub use ckb_smt::{SMTBuilder, SMT};
 pub use h256::H256;
 pub use merkle_proof::{CompiledMerkleProof, MerkleProof};
+#[cfg(not(feature = "trie"))]
 pub use tree::SparseMerkleTree;
+pub use tree::{BranchKey, BranchNode};
```

Comment:
Please move out `BranchKey` and `BranchNode` from `tree.rs` to something `branch.rs` and add feature attribute to `pub mod tree` as well.

Reply:
suggest to keep tree mod private:
https://github.com/code-monad/sparse-merkle-tree/pull/12

Reply:
> suggest to keep tree mod private: [code-monad#12](https://github.com/code-monad/sparse-merkle-tree/pull/12)

This PR has been merged.
IMO, keep `BranchKey`/`BranchNode` in `tree` may refer to a more stable & quick migration

Reply:
> > suggest to keep tree mod private: [code-monad#12](https://github.com/code-monad/sparse-merkle-tree/pull/12)
> 
> This PR has been merged. IMO, keep `BranchKey`/`BranchNode` in `tree` may refer to a more stable & simplified migration

Agreed